gdtchron/he.py

```diff
@@ -0,0 +1,671 @@
+            )
+        
+        # Solve for x using banded A and B
+        x = solve_banded((1, 1), ab, b)
+        
+    return x
+
+
+def profile_to_age(x, node_positions, radius, uth_molg, stop_distances):
+    """Calculate age based on final helium profile.
+
+    Calculate corrected and uncorrected age based on final He profile. Also
+    calculates normalized He profile and node positions.
+
+    Parameters
+    ----------
+    x : NumPy array of floats
+        Matrix x (mol * micrometers / g) solved for using Equation 21 from 
+        Ketcham (2005). 
+        Equivalent to the concentration times the node position. 
+        In Ketcham (2005), this variable is referred to as u, but x is used here
+        to distinguish this variable from the uranium-related variables.
+    node_positions : NumPy array of floats
+        Radial positions of each modeled node (micrometers)
+    radius : float
+        Radius of the grain (micrometers)
+    uth_molg : tuple
+        u238_molg : float
+            The concentration of U238 (mol / g),
+        u235_molg : float
+            The concentration of U235 (mol / g)
+        th_molg : float
+            The concentration of Th (mol / g)
+    stop_distances : NumPy array of floats
+        Stopping distances (micrometers) for (in order) U238, U235, and Th232
+
+    Returns
+    -------
+    age_corrected : float
+        Corrected (U-Th) / He age
+    age_uncorrected : Float
+        (U-Th) / He age without alpha correction
+    he_molg : float
+        Total amount (mol / g) of He within the modeled crystal.
+    position_normalized : NumPy array of floats
+        Radial positions of each modeled node, normalized so that the radius has
+        a radial position of 1
+    v_normalized : NumPy array of floats
+        Radial profile of He, normalized so that the position with the highest
+        concentration has a value of 1
+
+    """
+    he_molg, v = sum_he_shells(x, node_positions, radius)
+    
+    u238_molg, u235_molg, th_molg = uth_molg
+    
+    # Because alpha ejection modeled, model age is an "uncorrected" age.
+    age_uncorrected = calc_age(he_molg, u238_molg, u235_molg, th_molg)
+
+    # "Corrected" age uses alpha-adjusted U-Th values
+    
+    # Get array of correction values (238,235,232)
+    tau = alpha_correction(stop_distances, radius)
+    
+    # Correct U and Th accordingly
+    u238_corr = u238_molg * tau[0]
+    u235_corr = u235_molg * tau[1]
+    th_corr = th_molg * tau[2]
+    
+    age_corrected = calc_age(he_molg, u238_corr, u235_corr, th_corr)
+    
+    # Make diffusional profile
+    v_normalized = v / np.max(v)
+    position_normalized = node_positions / radius
+    
+    return (age_corrected, 
+            age_uncorrected, 
+            he_molg,
+            position_normalized,
+            v_normalized)
+
+
+def forward_model_he(temps, time_interval, system, u, th, radius, nodes=513,
+                  initial_x=None, return_all=True):
```

Comment:
One thing where I am noticing we don't have parity between `forward_model_he` and `forward_model_aft` - `he` asks for a time_interval and `aft` asks for a list of timesteps. My gut feeling is that we should have time fed into both the same way, and I don't think there's any reason one can't be converted to the other with a little tweaking. 

Maybe we should adopt the `aft` approach and have it be an array of times? We also should make the units consistent across inputs and outputs (so if we're outputting ages in Ma, inputs should also be in Ma rather than years BP).

My thinking here is that if someone is working in a Jupyter Notebook with a particular t-T path, they shouldn't have to reformat the path in order to feed it into `aft` or `he`.